driver/rke/rke_driver.go

```diff
@@ -0,0 +1,93 @@
+package rke
+
+import (
+	"io/ioutil"
+
+	generic "github.com/rancher/kontainer-engine/driver"
+	"github.com/rancher/rke/cmd"
+)
+
+// Driver is the struct of rke driver
+type Driver struct {
+	// The string representation of Config Yaml
+	ConfigYaml string
+	// Kubernetes master endpoint
+	Endpoint string
+	// Root certificates
+	RootCA string
+	// Client certificates
+	ClientCert string
+	// Client key
+	ClientKey string
+}
+
+// GetDriverCreateOptions returns create flags for rke driver
+func (d *Driver) GetDriverCreateOptions() (*generic.DriverFlags, error) {
+	driverFlag := generic.DriverFlags{
+		Options: make(map[string]*generic.Flag),
+	}
+	driverFlag.Options["config-file-path"] = &generic.Flag{
+		Type:  generic.StringType,
+		Usage: "the path to the config file ",
+	}
+	return &driverFlag, nil
+}
+
+// GetDriverUpdateOptions returns update flags for rke driver
+func (d *Driver) GetDriverUpdateOptions() (*generic.DriverFlags, error) {
+	// todo: rke doesn't have update
+	return nil, nil
+}
+
+// SetDriverOptions sets the drivers options to rke driver
+func (d *Driver) SetDriverOptions(driverOptions *generic.DriverOptions) error {
+	// first look up the file path then look up raw rkeConfig
+	if path, ok := driverOptions.StringOptions["config-file-path"]; ok {
+		data, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		d.ConfigYaml = string(data)
+		return nil
+	}
+	d.ConfigYaml = driverOptions.StringOptions["rkeConfig"]
+	return nil
+}
+
+// Create creates the rke cluster
+func (d *Driver) Create() error {
+	APIURL, caCrt, clientCert, clientKey, err := cmd.ClusterUp(d.ConfigYaml, "x509")
```

Comment:
@StrongMonkey should we add the `x509` to the driver config, or in rke's config yaml?, currently the auth type is passed as a cli flag.

Reply:
yeah, we can pass it as a cli flag and config